### PR TITLE
Adding more Numpy data types

### PR DESCRIPTION
**Pull request by szalpal**
Signed-off-by: szalpal <mszolucha@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It fixes a bug - missing numpy data types

Bug discovered in #1958 

**JIRA TASK**: *[Use DALI-XXXX or NA]*

### CODE REVIEW

**Comment by klecki** [comment]
Is there a format string for `bool`? Is `bool` possible/valid here?

**Comment by szalpal** [comment]
There is, but C++ doesn't really map to fixed-length type for `bool`, so I didn't include it. Should I?

**Comment by szalpal** [comment]
!build

**Comment by klecki** [comment]
> There is, but C++ doesn't really map to fixed-length type for `bool`, so I didn't include it. Should I?

Yes, apparently it's implementation defined.

**Comment by JanuszL** [comment]
Can you add a regression test for that?

**Comment by szalpal** [comment]
!build

File: dali/operators/reader/loader/numpy_loader.cc [hunk 1]
```diff
@@ -24,16 +24,18 @@
 
 namespace dali {
 TypeInfo TypeFromNumpyStr(const std::string &format) {
```
Comment (JanuszL) [inline dali/operators/reader/loader/numpy_loader.cc:26]: Why not to reuse `TypeFromFormatStr` from https://github.com/NVIDIA/DALI/blob/master/dali/util/pybind.h#L103?

**Pull request merged**